Perez diet

CBjerke opened the pull request:
diet score (perez) created

wyusuf068 commented on R/diet.R:
```diff
@@ -0,0 +1,132 @@
+                                            FVCDVEG)
+    
+    # Nested function: maximum total fruit and vegetables = 8
+    derive_max_fruitveg <- function(FVCDFRU, FVCDSAL, FVCDPOT, FVCDCAR, 
```
`derive_max_fruitveg()` should be a function with one parameter: `total_fruitveg`

CBjerke replied:
oops...yup

wyusuf068 reviewed (changes requested):
Overall looks good. See comment, and don't forget to update `variables.csv` and `variable_details.csv`.

yulric commented on R/diet.R:
```diff
@@ -0,0 +1,133 @@
+    derive_max_fruitveg <- function(FVCDFRU, FVCDSAL, FVCDPOT, FVCDCAR, 
+                                    FVCDVEG) {
+      max_fruitveg <-
+        if_else2(total_fruitveg >8, 8,
```
The check for whether total_fruitveg is NA should happen before the check whether its greater than 8.

yulric commented on R/diet.R:
```diff
@@ -0,0 +1,133 @@
+    
+    # Nested function: high potato intake flag
+    potato_fun <- function(FVCDPOT, DHH_SEX) {
+      FVCDPOT_high <-
```
This code should be split up into 2 parts to avoid checking for DHH_SEX twice. The first part get daily limit to use depending on the sex and the second part sets the high flag based on that calculated daily limit.

yulric commented on R/diet.R:
```diff
@@ -0,0 +1,133 @@
+    # Nested function: no carrot intake flag
+    carrot_fun <- function(FVCDCAR) {
+      FVCDCAR_nil <-
+        if_else2(FVCDCAR==0, 1,
```
Check for whether its not NA should happen before check for 0.

yulric commented on R/diet.R:
```diff
@@ -0,0 +1,133 @@
+    juice_fun <- function(FVCDJUI) {
+      FVCDJUI_high <-
+        if_else2(FVCDJUI <=1, 0,
+                if_else2(!is.na(FVCDJUI), FVCDJUI - 1, NA))
```
Check for whether its not NA should happen before the check for <= 1.

yulric reviewed (changes requested):
1. Does it make sense that the diet function is returning 2 variables? diet_raw_score and diet_score.
2. Similar to the smoke simple branch I don't think the nested functions are necessary.

CBjerke commented:
> 1. Does it make sense that the diet function is returning 2 variables? diet_raw_score and diet_score.
> 2. Similar to the smoke simple branch I don't think the nested functions are necessary.

I've removed nested functions and changed code to just return diet_score

CBjerke commented:
> Overall looks good. See comment, and don't forget to update `variables.csv` and `variable_details.csv`.

@wyusuf068 which branch should I work off to update the variable and variable_details? It seems as if the versions in the 'Perez diet' branch only contain variables from the PHIAT YLL project.

wyusuf068 commented:
> > Overall looks good. See comment, and don't forget to update `variables.csv` and `variable_details.csv`.
> 
> @wyusuf068 which branch should I work off to update the variable and variable_details? It seems as if the versions in the 'Perez diet' branch only contain variables from the PHIAT YLL project. Also, not sure how the variables sheet should be updated (since the base variables are already there)...or do you want the intermediate variables added?

You can update variables and variable_details.csv in this branch by adding to the bottom of each sheet. To add to the two sheets you would just have to specify the final derived variable with variableStart being the base variables (DerivedVar::[FVCDFRU, FVCDSAL,...]) and the recTo being name of the function (Func::diet_score_fun).
